framework/Models/PostProcessors/ParetoFrontierPostProcessor.py

```diff
@@ -126,40 +113,24 @@ def run(self, inputIn):
     inData = self.inputToInternal(inputIn)
     data = inData.asDataset()
 
-    if self.invCost:
-      data[self.costID] = (-1.) * data[self.costID]
-    if self.invValue:
-      data[self.valueID] = (-1.) * data[self.valueID]
-
-    sortedData = data.sortby(self.costID)
-    coordinates = np.zeros(1,dtype=int)
-    for index,elem in enumerate(sortedData[self.costID].values):
-      if (index>1) and (sortedData[self.valueID].values[index]>sortedData[self.valueID].values[coordinates[-1]]):
-        # the point at index is part of the pareto frontier
-        coordinates = np.append(coordinates,index)
-
-    selection = sortedData.isel(RAVEN_sample_ID=coordinates)
-
-    if self.invCost:
-      selection[self.costID] = (-1.) * selection[self.costID]
-    if self.invValue:
-      selection[self.valueID] = (-1.) * selection[self.valueID]
-
-    if self.valueLimit is not None:
-      if self.valueLimitType=="upper":
-        selection = selection.where(selection[self.valueID]<=self.valueLimit)
-      else:
-        selection = selection.where(selection[self.valueID]>=self.valueLimit)
-    if self.costLimit is not None:
-      if self.costLimitType=="upper":
-        selection = selection.where(selection[self.costID]<=self.costLimit)
-      else:
-        selection = selection.where(selection[self.costID]>=self.costLimit)
+    dataTemp = data[list(self.objectives.keys())]
+    for index,obj in enumerate(self.objectives.keys()):
+      if self.objectives[obj]['goal']=='max':
+        dataTemp[obj] = (-1.) * dataTemp[obj]
+
+    paretoFrontMask = frontUtils.nonDominatedFrontier(np.transpose(dataTemp.to_array().values), returnMask=False)
+    selection = data.isel(RAVEN_sample_ID=np.array(paretoFrontMask))
+
+    for obj in self.objectives.keys():
+      if 'upperLimit' in self.objectives[obj].keys():
```

Comment:
If you make changes according to previous comments, this line can be changed: 
if self.objectives[obj]['upperLimit']:

Reply:
yup, code is cleaner